### DISCUSSION TITLE
Can't register new keys with webauthn due to Content-Security-Policy presumably blocking javascript

### DISCUSSION THREAD

**Discussion by PlexSheep** [authelia/authelia#5932] [Troubleshooting] [answered]
**Disclaimer: I am using authelia:v4.38.0-beta2**

When I try to register a Security Key, the process fails, stating that the request is malformed or my browser not supported. I have just set up authelia today for the first time, so I have a feeling that I'm doing something wrong.

![Screenshot_20230830_215240](https://github.com/authelia/authelia/assets/58274330/5b828c27-3539-4e42-b770-7b066e1c0c39)

![Screenshot_20230830_215356](https://github.com/authelia/authelia/assets/58274330/f9d76f75-ad68-460a-820a-2073f8a5d103)

Translation: Content-Security-Policy: The Settings of the site have blocked the loading of a ressource ("default-src")

The CSP, as cURL receives it: `content-security-policy: default-src 'self'; frame-src 'none'; object-src 'none'; style-src 'self' 'nonce-7BBy34AptVpAMfqAE0FbjfDq3xHH6sfZ'; frame-ancestors 'none'; base-uri 'self'`

Tested on both:
- Firefox 116.0.2 (64-Bit)
- Chromium Version 115.0.5790.170 (Official Build Fedora Project (64-Bit)

With two seperate Hardware Keys.

I am using Docker for hosting, here are the relevant files:
(GitHub does not allow files with the yml extension for some reason)
[configuration.yml.txt](https://github.com/authelia/authelia/files/12479759/configuration.yml.txt)
[docker-compose.yml.txt](https://github.com/authelia/authelia/files/12479760/docker-compose.yml.txt)

# Actual Question
Is there some config I need to add, so that the script can be served under the CSP? I've tried to tinker with the `csp_template`, but without success. Or is it really a bug in the Project? Is the script being blocked by the CSP even the reason that this happens?

**Comment by james-d-elliott** [comment, marked as answer]
https://github.com/authelia/authelia/discussions/5710#discussioncomment-6780204
